### ISSUE TITLE
Filter by currently available

### ISSUE THREAD

**Issue by Thortok** [Malin001/Legilimens-Hogwarts-Legacy-cpp#2]
Not a complete list by any means, just what I worked on tonight, but here's some data on some I was not able to get and my note about what would be required to get it. If you haven't added the feature by the time I get to play some more I'll post any more data I collect.

Level recommendation is just a suggestion and probably not strictly necessary to code into the program, more for my personal notes, but up to you if you want to consider it.

Hogsmeade	
FGP #47	Loom
CC#2	Alohomora 3
CC#3	Alohomora 3
CC#4	Alohomora 3
CC#5	A Basis for Blackmail
The Bell Tower Wing	
CC#6	Depulso
The Grand Staircase	
FGP #6	In the Shadow of the Study
FGP #22	Headmaster's Office
FGP #23	Headmaster's Office
CC #5	Headmaster's Office
FGP #24	Headmaster's Office
CC #6	Headmaster's Office
CC #7	Headmaster's Office
The Library Annex	
CC #6	Depulso
The South Wing	
CC #4	Alohomora 3
FGP #14	Alohomora 3
FGP #19	Depulso
Vivariums	Vivariums
Clagmar Coast	
MT #1	Depulso
MT #5	Flipendo
IF #1	Level 37
CC #4	Level 30
CC #5	Level 30
Coastal Cavern	
IF #1	Level 30

**Comment by Thortok** [comment]
Got to play on my Slytherin a little bit this morning. Here's my notes:

Hogsmeade	
DM #3	Alohomora 2
DM #4	Alohomora 2
FGP #47	Loom
CC# 2	Alohomora 3
CC# 3	Alohomora 3
CC# 4	Alohomora 3
CC# 5	A Basis for Blackmail
The Astronomy Wing	
DM #1	Alohomora 2
CC #2	Alohomora 2
FGP #4	Alohomora 2
FGP #13	Alohomora 2
FGP #20	Alohomora 2
CC #6	Alohomora 2
FGP #2	Alohomora 2
The Bell Tower Wing	
CC #1	Alohomora 2
DM #2	Alohomora 2

**Comment by Thortok** [comment]
[Hogwarts Legacy.xlsx](https://github.com/Malin001/Legilimens-Hogwarts-Legacy-cpp/files/10915690/Hogwarts.Legacy.xlsx)

My current notes on what's blocking my collections.

**Comment by Thortok** [comment]
[Hogwarts Legacy.xlsx](https://github.com/Malin001/Legilimens-Hogwarts-Legacy-cpp/files/10921141/Hogwarts.Legacy.xlsx)
Another play session, notes updated.

**Comment by Thortok** [comment]
[Hogwarts Legacy.xlsx](https://github.com/Malin001/Legilimens-Hogwarts-Legacy-cpp/files/10925645/Hogwarts.Legacy.xlsx)
More notes

**Comment by Malin001** [comment]
Thanks so much for this! I'm currently working on detecting which quests are done, which levels of alohamora are unlocked, etc.
Once I have that done it should be relatively easy (with your notes to help) to add the obtainable filter. Ideally I'll finish it this weekend if I have enough free time.

**Comment by Thortok** [comment]
I definitely do plan to do a lot of playing this weekend and should collect more data. I've decided to halt my progression through the storyline on all 4 characters and collect as much as possible before continuing quests.

Might I also suggest rearranging the zones in a more logical order such as the order they'd naturally be encountered when following the main storyline? Right now it seems like hogsmeade, hogwarts, then alphabetically.

I also suggest merging butterflies and daedalean keys (or however you spell it) with their relevant zones. The vivariums could even go in the 'astronomy wing' list as well, technically.

As always, any way I can help, let me know. Thanks again for the great tool!

**Comment by Thortok** [comment]
Rather than spam you with updates and attachments and such here's a link which should reflect live any updates I make as I play and update it.

https://1drv.ms/x/s!Aky6Yv3AUCIqgpIHJ4h1H8U1jMf3GA?e=IYexyR

**Comment by Malin001** [comment]
Yeah I've debated about what the "best" order is for the various regions, and figured that keeping the 3 main regions (Hogwarts/Hogsmeade/The Highlands) was probably a good idea, and alphabetically should in theory make each region easy to locate. And then I added butterflies/daedalian keys/finishing touches afterwards and that kinda messed it up lol. As for merging butterflies and keys into their regions, it makes more sense to do that now that I've added the ability to sort by type, but on the other hand each region currently contains exactly the items that show up when hovering over that region on the in-game map, so you could argue that either way is better. So maybe I'll make those changes, or maybe I won't, idk.

As for ways you can help, just updating that excel sheet is perfect. Unless some new errors/bugs appear, this is the only feature that I'll add in the next update, so it's getting my full focus this week.

Thanks for your help!

**Comment by Thortok** [comment]
So I noticed collecting was giving me a lot of xp and I was worried I'd level up to max level before unlocking the talents I want. My Slytherin has stopped collecting in order to get the dark arts, and my Gryffindor has stopped collecting in order to finish the professor assignments. Once they get the talents they want they'll go back to collecting. And my Ravenclaw and Hufflepuff already have the talents they want, so they are still collecting as well. I'll let you know if/when they go all the way through and their blocked list is 'complete' and they have to go back to questing to progress. You'll probably be done by then anyway, though.

Note my sheet is just my personal indication of what's blocking that character from getting that thing. If the character was able to collect that thing, I leave it blank, and if all four characters have collected the thing then I delete the line. So this information will only be partially helpful to you at best for creating filters for everyone. Still, better than nothing, and I hope it helps a little at least!

**Comment by Thortok** [comment]
My Ravenclaw has completed collections and is now moving on with doing quests and continuing the storyline. You may want to grab a copy of the sheet now as it will be deleted/erased as my progress continues and I unlock and collect stuff. Again, it's not the full amount of information necessary to collect any given thing, just what is blocking my specific character(s) from getting it. But it still might be of some help regardless.
